src/ir/daphneir/Canonicalize.cpp

```diff
@@ -249,6 +249,25 @@ mlir::LogicalResult mlir::daphne::SparsityOp::canonicalize(mlir::daphne::Sparsit
 mlir::LogicalResult mlir::daphne::EwAddOp::canonicalize(mlir::daphne::EwAddOp op, PatternRewriter &rewriter) {
     mlir::Value lhs = op.getLhs();
     mlir::Value rhs = op.getRhs();
+    // This will check for the fill operation on the left hand side to push down the arithmetic inside
+    // of it
+    auto lhsFill = lhs.getDefiningOp<mlir::daphne::FillOp>();
+    if (lhsFill) {
+        auto lhsOp = lhs.getDefiningOp();
+        auto fillValue = lhsOp->getOperand(0);
+        auto width = lhsOp->getOperand(1);
+        auto height = lhsOp->getOperand(2);
+        const bool rhsIsSca = !llvm::isa<mlir::daphne::MatrixType, mlir::daphne::FrameType>(rhs.getType());
```

Comment:
I recommend relying on `CompilerUtils::isScaType()` to check if an `mlir::Value` is a scalar. Beyond matrices and frames, there are more types that are not scalars (e.g., lists, columns (since recently), and a few more technical ones).

Reply:
Thanks.

Further down in the existing code there are 2 more checks

```
  292 ▎   ▎   const bool lhsIsSca = !llvm::isa<mlir::daphne::MatrixType, mlir::daphne::FrameType>(lhs.getType());
  293 ▎   ▎   const bool rhsIsSca = !llvm::isa<mlir::daphne::MatrixType, mlir::daphne::FrameType>(rhs.getType());

```

Should I rewrite those as well while I am in this function?